streams/src/main/java/org/apache/kafka/streams/state/StateSerdes.java

```diff
@@ -73,6 +101,11 @@ public StateSerdes(String stateName,
         this.valueSerde = valueSerde;
     }
 
+    public StateSerdes<K, V> setApplicationId(final String applicationId) {
```

Comment:
I'm not sure about this. It seems to be used only in the where we are using `WindowStoreUtils.INNER_SERDES`. I think i'd prefer to just create a new instance for each of those cases and do away with the `static final`